### ISSUE TITLE
Support inclusions of complex object graphs

### ISSUE THREAD

**Issue by remmeier** [katharsis-project/katharsis-framework#364]
complex object graphs, e.g. with parents and children and other cyclic structures, are currently not properly supported. An attribute can only be included once, regardless of its position in the object graph. For example, include=parent.parent.children is currently cut off after the first parent. This limiation should be removed. 

Note that:
any implementation must make sure that the IncludeLookupSetter does not get trapped in a cycle. So instead of allowing to follow every attribute only once, it must enforce to follow every attribute per resource only once. As a consequence, IncludeLookupSetter must keep track of what he has already done.

Further note that:
we currently have two inclusion mechanisms: 

- `GET /tasks/?include[projects]=task&include[tasks]=comments`
- `GET /tasks/?include[projects]=task.comments`

for cyclic structures they do have different semantics, the former becoming recursive. A feature flag should to be introduced to disable the former and let the application decide whether to use it or not. By default it is enabled to maintain the current behavior.
